rsql-common/src/main/java/io/github/perplexhub/rsql/RSQLCommonSupport.java

```diff
@@ -37,6 +37,7 @@ public RSQLCommonSupport() {
 	public RSQLCommonSupport(Map<String, EntityManager> entityManagerMap) {
 		if (entityManagerMap != null) {
 			RSQLCommonSupport.entityManagerMap.putAll(entityManagerMap);
+			RSQLCommonSupport.managedTypeMap.clear();
```

Comment:
Should we clear managedTypeMap in constructor?

Reply:
I found the problem in my company project that _sometimes_ must clear Spring context in IT test. This fix address this case but I'm not sure this covers all the possible side effects of the static cache.

Reply:
I agree the managedTypeMap must be cleared in some test. I think the cache should be cleared by @BeforeEach/@BeforeAll between the tests.
In some production code, managedTypeMap would be assigned before calling the constructor, i.e. the change would clear the predefined mapping.

Reply:
Yes I do it manually on affected existing projects but it was not trivial to find the problem.

Reply:
```java
RSQLVisitorBase.setEntityManagerMap(getEntityManagerMap());
RSQLVisitorBase.setManagedTypeMap(getManagedTypeMap());
RSQLVisitorBase.setPropertyRemapping(getPropertyRemapping());
RSQLVisitorBase.setGlobalPropertyWhitelist(getPropertyWhitelist());
RSQLVisitorBase.setGlobalPropertyBlacklist(getPropertyBlacklist());
RSQLVisitorBase.setDefaultConversionService(getConversionService());	
```
We could implement above ```setXXX``` methods to reset and copy the caches.